From 53e6d889ac29336ba212a0d4a987455a85736fa8 Mon Sep 17 00:00:00 2001
From: Himanshu Agarwal <himanaga@codeaurora.org>
Date: Tue, 26 Dec 2017 17:21:28 +0530
Subject: qcacld-3.0: Calculate buf_len properly for extscan hotlist event
 buffer

Calculate buf_len properly for extscan hotlist event buffer in
wma_extscan_hotlist_match_event_handler()

Change-Id: I923f0d86d83902539a7d3a7b8ea8322dcb83ba00
CRs-Fixed: 2161977
---
 core/wma/src/wma_scan_roam.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/wma/src/wma_scan_roam.c b/core/wma/src/wma_scan_roam.c
index 94a55cf..6e1cfad 100644
--- a/core/wma/src/wma_scan_roam.c
+++ b/core/wma/src/wma_scan_roam.c
@@ -4402,7 +4402,7 @@ int wma_extscan_hotlist_match_event_handler(void *handle,
 	}
 
 	buf_len = sizeof(wmi_extscan_hotlist_match_event_fixed_param) +
-		  (4 * sizeof(uint32_t)) +
+		  WMI_TLV_HDR_SIZE +
 		  (numap * sizeof(wmi_extscan_wlan_descriptor));
 
 	if (buf_len > len) {
-- 
cgit v1.1

